assets/js/common/Filter/Filter.jsx

```diff
@@ -14,26 +14,37 @@ const getLabel = (value, placeholder) =>
 /**
  * Filter component
  *
- * @param {string[]} props.options List of options to filter, e.g. ['Option 1', 'Option 2']
+ * @param {string[]|[string, string][]} props.options List of options to filter, e.g. ['Option 1', 'Option 2']
  * @param {string} props.title Title of the filter. It will be displayed in the button when the filter is empty
- * @param {string[]} props.value Selected options. Default is an empty array
+ * @param {string|string[]} props.value Selected options. Default is an empty array
  * @param {function} props.onChange Function to call when the selected options change
  */
 function Filter({ options, title, value = [], onChange }) {
   const ref = useRef();
   const [open, setOpen] = useState(false);
   const [query, setQuery] = useState('');
 
-  const filteredOptions = options
+  const labeledOptions = options
     .filter((option) => option !== undefined && option !== null)
-    .filter((option) => option.toLowerCase().includes(query.toLowerCase()));
+    .map((option) => (typeof option === 'string' ? [option, option] : option));
+
+  const filteredOptions = labeledOptions.filter((option) =>
+    option[0].toLowerCase().includes(query.toLowerCase())
+  );
+
+  const parsedValue = typeof value === 'string' ? [value] : value;
+
+  const selectedLabels = parsedValue.reduce((acc, key) => {
+    const e = labeledOptions.find(([optionKey]) => optionKey === key);
```

Comment:
```suggestion
    const element = labeledOptions.find(([optionKey]) => optionKey === key);
```

Reply:
Done in https://github.com/trento-project/web/pull/2892/commits/d039a74a8a8575e808a6b48f36b179d6aaae46e4